internal/service/elasticsearch/domain.go

```diff
@@ -639,6 +700,24 @@ func resourceDomainCreate(d *schema.ResourceData, meta interface{}) error {
 
 	log.Printf("[DEBUG] Elasticsearch domain %q created", d.Id())
 
+	log.Printf("[DEBUG] Modifying config for Elasticsearch domain %q", d.Id())
```

Comment:
We only need to call `UpdateElasticsearchDomainConfig` if there are auto-tune options.
As tests are passing, calling this API in other cases (effectively no update) must not be causing problems.
For completeness, you could move the update config logic inside the `if`:

```go
if v, ok := d.GetOk("auto_tune_options"); ok && len(v.([]interface{})) > 0 {
    log.Printf("[DEBUG] Modifying config for Elasticsearch domain %q", d.Id())

    ...

    log.Printf("[DEBUG] Config for Elasticsearch domain %q modified", d.Id())
}
```

Reply:
Change adopted!